crates/revm/src/processor.rs

```diff
@@ -234,15 +249,14 @@ impl<'a> EVMProcessor<'a> {
         total_difficulty: U256,
         senders: Option<Vec<Address>>,
     ) -> Result<(Vec<Receipt>, u64), BlockExecutionError> {
+        self.init_env(&block.header, total_difficulty);
```

Comment:
does this need to happen even if the block is empty? if not, should prob move back to where it was before